testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -1214,6 +1221,152 @@ impl BitcoinRegtestController {
         Some(tx)
     }
 
+    #[cfg(not(test))]
+    fn build_peg_out_request_tx(
+        &mut self,
+        _epoch_id: StacksEpochId,
+        _payload: PegOutRequestOp,
+        _signer: &mut BurnchainOpSigner,
+    ) -> Option<Transaction> {
+        unimplemented!()
+    }
+
+    #[cfg(test)]
+    fn build_peg_out_request_tx(
+        &mut self,
+        epoch_id: StacksEpochId,
+        payload: PegOutRequestOp,
+        signer: &mut BurnchainOpSigner,
+    ) -> Option<Transaction> {
+        let public_key = signer.get_public_key();
+        let max_tx_size = 230;
+        let dust_amount = 10000;
+
+        let output_amt = DUST_UTXO_LIMIT
+            + max_tx_size * self.config.burnchain.satoshis_per_byte
+            + payload.amount
+            + payload.fulfillment_fee;
+        let (mut tx, mut utxos) =
+            self.prepare_tx(epoch_id, &public_key, output_amt, None, None, 0)?;
+
+        let op_bytes = {
+            let mut bytes = self.config.burnchain.magic_bytes.as_bytes().to_vec();
+            bytes.push(Opcodes::PegOutRequest as u8);
+            bytes.extend_from_slice(&payload.amount.to_be_bytes());
+            bytes.extend_from_slice(payload.signature.as_bytes());
+            bytes
+        };
+
+        let amount_and_signature_output = TxOut {
+            value: 0,
+            script_pubkey: Builder::new()
+                .push_opcode(opcodes::All::OP_RETURN)
+                .push_slice(&op_bytes)
+                .into_script(),
+        };
+
+        tx.output = vec![amount_and_signature_output];
+        tx.output
+            .push(payload.recipient.to_bitcoin_tx_out(dust_amount));
```

Comment:
I think this is a bug, and an oversight in your integration test.  The peg-out amount for the first non-data output -- the payment to the peg wallet -- should be `payload.amount`, not `dust_amount`.